### PR TITLE
container: T5955: add uid/gid settings

### PR DESCRIPTION
**Pull request by ishioni**
## Change Summary
Adds the ability to set the uid/gid of the main process running inside the container

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5955

## How to test
```
set container name uidtest uid 1002
set container name uidtest gid 100
set container name uidtest image debian
...
sudo podman exec uidtest id
# Results in uid=1002(vyos) gid=100(users) groups=100(users)
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -316,6 +316,30 @@
...
           </leafNode>
+          <leafNode name="uid">
+            <properties>
+              <help>User id this container will run as</help>
+              <valueHelp>
+                <format>u32:0-65535</format>
+                <description>User id this container will run as</description>
+              </valueHelp>
+              <constraint>
+                <regex>(0|[1-9]\d{0,4}|[1-5]\d{4}|6[0-5]\d{3}|655[0-3][0-5])</regex>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: Is there a particular reason to use this complex regex over our range validation?

`<validator name="numeric" argument="--range 1-65535"/>`

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/container.xml.in [hunk 2]
```diff
@@ -316,6 +316,30 @@
             </properties>
             <defaultValue>on-failure</defaultValue>
           </leafNode>
+          <leafNode name="uid">
+            <properties>
+              <help>User id this container will run as</help>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: Glibc calls id `User ID` can you please use uppercase ID?

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/container.xml.in (hunk 1 above)
Comment (ishioni) [reply interface-definitions/container.xml.in]: I'm still pretty new to the codebase and didn't notice that option. Fixed

File: interface-definitions/container.xml.in (hunk 2 above)
Comment (ishioni) [reply interface-definitions/container.xml.in]: Done, also for Group

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
```
cpo@LR1.wue3:~$ /usr/libexec/vyos/tests/smoke/cli/test_container.py
test_basic (__main__.TestContainer.test_basic) ... ok
test_dual_stack_network (__main__.TestContainer.test_dual_stack_network) ... ok
test_ipv4_network (__main__.TestContainer.test_ipv4_network) ... ok
test_ipv6_network (__main__.TestContainer.test_ipv6_network) ... ok
test_uid_gid (__main__.TestContainer.test_01uid_gid) ... ok

----------------------------------------------------------------------
Ran 5 tests in 103.232s

OK
```

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Comment by ishioni** [comment]
@c-po Should i make a PR for docs?

**Comment by c-po** [comment]
Please do so

**Pull request merged**